File: java/program/Mesin.java
class Mesin {
    private String tipe;
    private String kapasitas;
    private String silinder;
    private String bahanBakar;

    public Mesin(){
        this.tipe = "";
        this.kapasitas = "";
        this.silinder = "";
        this.bahanBakar = "";
    }

    public Mesin(String tipe, String kapasitas, String silinder, String bahanBakar) {
        this.tipe = tipe;
        this.kapasitas = kapasitas;
        this.silinder = silinder;
        this.bahanBakar = bahanBakar;
    }

    public String getTipe() {
        return tipe;
    }

    public String getKapasitas() {
        return kapasitas;
    }

    public String getSilinder() {
        return silinder;
    }

    public String getBahanBakar() {
        return bahanBakar;
    }

    public void setTipe(String tipe) {
        this.tipe = tipe;
    }

    public void setKapasitas(String kapasitas) {
        this.kapasitas = kapasitas;
    }

    public void setSilinder(String silinder) {
        this.silinder = silinder;
    }

    public void setBahanBakar(String bahanBakar) {
        this.bahanBakar = bahanBakar;
    }
}

File: cpp/program/mesin.cpp
#ifndef MESIN_CPP
#define MESIN_CPP

#include <bits/stdc++.h>
using namespace std;

class Mesin{
    private:
        string tipe;
        string kapasitas;
        string silinder;
        string bahanBakar;
    public:
        Mesin(){
            tipe = "";
            kapasitas = "";
            silinder = "";
            bahanBakar = "";
        }

        Mesin(string tipe, string kapasitas, string silinder, string bahanBakar){
            this->tipe = tipe;
            this->kapasitas = kapasitas;
            this->silinder = silinder;
            this->bahanBakar = bahanBakar;
        }

        string getTipe() const{
            return tipe;
        }

        string getKapasitas() const{
            return kapasitas;
        }

        string getSilinder() const{
            return silinder;
        }

        string getBahanBakar() const{
            return bahanBakar;
        }

        void setTipe(string tipe){
            this->tipe = tipe;
        }

        void setKapasitas(string kapasitas){
            this->kapasitas = kapasitas;
        }

        void setSilinder(string silinder){
            this->silinder = silinder;
        }

        void setBahanBakar(string bahanBakar){
            this->bahanBakar = bahanBakar;
        }

        ~Mesin(){
            // Destructor
        }
};

#endif